### PR TITLE
MQTT Device Tracker schemas

### PR DESCRIPTION
**Pull request by PeteBa**
## Proposed change
Triggered by a request [here](https://github.com/home-assistant/home-assistant.io/pull/15427#issuecomment-764912091) to more fully document the schemas associated with the mqtt device_tracker.

@emontnemery , I just wanted to check-in with you before heading down the wrong track on this PR.

The mqtt device_tracker has two schemas: YAML schema is documented in `source/_integrations/device_tracker.mqtt.markdown` and the DISCOVERY schema is documented in `source/_docs/mqtt/discovery.markdown`. This is inline with other entities that separate the yaml versus discovery documentation.

I took a look at the light and vacuum examples but they seem to be a bit different. They have multiple YAML schemas and it makes sense to document those together. But that is not the case with device_tracker i.e. you cant use the DISCOVERY schema via yaml.

Let me know your thoughts as I'm keen to get this right and I'm happy to do the work.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

Link to original documentation update #15427
- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by emontnemery** [comment]
Note that the documentation in `source/_docs/mqtt/discovery.markdown` is not complete, it's just examples.

I think it's fine to document both schemas in `source/_integrations/device_tracker.mqtt.markdown`, just mention in the introduction that the old schema only supports setup through `configuration.yaml` while the new schema only supports setup trough MQTT discovery.

As an example, MQTT tag scanner only supports setup through MQTT discovery: https://www.home-assistant.io/integrations/tag.mqtt/

**Comment by PeteBa** [comment]
@emontnemery, appreciate if you could take a look at this to see if inline with expectation. I agree that this provides fuller documentation that should open up the discovery side of things. Happy to tweak and extend as needed.

Also, does anyone know how to see fully formatted {% Configuration %} sections. They dont seem to be rendered in VS Code markdown extension nor in github file view.

**Comment by LordMike** [comment]
Chiming in here - the [current example on the Discovery page](https://www.home-assistant.io/docs/mqtt/discovery/#presence-detection-device-tracker) for the device_tracker, posts this example - but HASS doesn't accept it.

Example:

```json
{
  "name":"Paulus",
  "state_topic": "homeassistant/device_tracker/paulus/state",
  "payload_home": "home",
  "payload_not_home": "not_home",
  "source_type": "bluetooth",
 }
```

HASS (2021.1.5) logs:

```
2021-02-04 22:12:18 WARNING (MainThread) [homeassistant.components.mqtt.discovery] Unable to parse JSON my_tracker: '{
  "name":"Paulus",
  "state_topic": "homeassistant/device_tracker/paulus/state",
  "payload_home": "home",
  "payload_not_home": "not_home",
  "source_type": "bluetooth",
 }'
```

EDIT: Found it - there's an extra "," at the last property.

**Comment by PeteBa** [comment]
@LordMike , although not specific to this PR, I do appreciate your feedback on the content of a previous PR #15427. I have raised #16417 for the corrections. Thanks.

**Comment by emontnemery** [comment]
@PeteBa To see how tings turn out, either build locally, or have a look at the CI build:
image

**Comment by PeteBa** [comment]
@emontnemery , thanks for the additional fixes. I have reviewed the build version and the `{% configuration %}` formatting looks good. I have seen one more tweak needed on line 112 that should substitute "device tracker" for "device trigger". I just need to figure out how to sync your commits back to my local branch so I can make this update (#advanced-gitfu 8) !

File: source/_integrations/device_tracker.mqtt.markdown [hunk 1]
```diff
@@ -64,13 +64,154 @@ device_tracker:
...
+  description: When `availability` is configured, this controls the conditions needed to set the entity to `available`. Valid entries are `all`, `any`, and `latest`. If set to `all`, `payload_available` must be received on all configured availability topics before the entity is marked as online. If set to `any`, `payload_available` must be received on at least one configured availability topic before the entity is marked as online. If set to `latest`, the last `payload_available` or `payload_not_available` received on any configured availability topic controls the availability.
+  required: false
+  type: string
+  default: latest
+availability_topic:
+  description: The MQTT topic subscribed to receive availability (online/offline) updates. Must not be used together with `availability`.
+  required: false
+  type: string
+device:
+  description: "Information about the device this device trigger is a part of that ties it into the [device registry](https://developers.home-assistant.io/docs/en/device_registry_index.html)."
```
Comment (PeteBa) [inline source/_integrations/device_tracker.mqtt.markdown]:
```suggestion
  description: "Information about the device this device tracker is a part of that ties it into the [device registry](https://developers.home-assistant.io/docs/en/device_registry_index.html)."
```

**Comment by PeteBa** [comment]
OK. Found a workaround to get the "trigger->tracker" tweak made. I'm good with the PR now but let me know if you have any additional changes required.

**Review by emontnemery — APPROVED** [review]

**Pull request merged**